apps/site/assets/ts/models/alert.ts

```diff
@@ -19,6 +20,93 @@ export const alertsByStop = (alerts: Alert[], stopId: StopId): Alert[] =>
       !!entities.stop && entities.stop!.some((id: StopId) => id === stopId)
   );
 
+const hasEffect = (alerts: Alert[], effect: string): boolean =>
+  some(alerts, alert => alert.effect === effect);
+
+export const isSuspension = (alerts: Alert[]): boolean =>
+  hasEffect(alerts, "suspension");
+
+export const isShuttleService = (alerts: Alert[]): boolean =>
+  hasEffect(alerts, "shuttle");
+
+export const isDetour = (alerts: Alert[]): boolean =>
+  hasEffect(alerts, "detour");
+
+export const alertsWithStop = (alerts: Alert[]): Alert[] =>
+  alerts.filter(
+    ({ informed_entity: entites }: Alert): boolean => !!entites.stop
+  );
+
+export const routeWideAlerts = (alerts: Alert[]): Alert[] =>
+  alerts.filter(({ informed_entity: { entities } }: Alert): boolean =>
+    entities.some(entity => !entity.stop && !entity.trip)
+  );
+
+// Can there be alerts that don't affect routes
+export const alertsByRoute = (alerts: Alert[]): { [key: string]: Alert[] } => {
+  return reduce(
+    alerts,
+    (acc, alert) => {
+      // create a map of every route id to the alert
+      const explodedAlerts = reduce(
+        alert.informed_entity.route,
+        (innerAcc, routeId) => {
+          return { ...innerAcc, [routeId]: [alert] };
+        },
+        {}
+      );
+
+      return mergeWith(acc, explodedAlerts, (objValue, srcValue) => {
+        if (isArray(objValue)) {
+          return objValue.concat(srcValue);
+        }
+        return srcValue;
+      });
+    },
+    {}
+  );
+};
+
+// This will only return alerts with direction id specified (aka affecting a single direction)
+export const alertsByDirectionId = (
+  alerts: Alert[]
+): { [key: number]: Alert[] } => {
+  return reduce(
+    alerts,
+    (acc, alert) => {
+      const explodedAlerts = reduce(
+        alert.informed_entity.direction_id,
+        (innerAcc, directionId) => {
+          // only include specified directions
+          if (directionId !== null) {
+            return { ...innerAcc, [directionId]: [alert] };
+          }
+          return innerAcc;
+        },
+        {}
+      );
+
+      return mergeWith(acc, explodedAlerts, (objValue, srcValue) => {
+        if (isArray(objValue)) {
+          return objValue.concat(srcValue);
+        }
+        return srcValue;
+      });
+    },
+    {}
+  );
+};
+
+export const alertsAffectingBothDirections = (alerts: Alert[]): Alert[] => {
+  return alerts.filter(alert => {
+    return (
+      alert.informed_entity.direction_id === null ||
+      alert.informed_entity.direction_id.length === 0 ||
+      alert.informed_entity.direction_id[0] === null
+    );
```

Comment:
question: would `direction_id` list ever look like `[0, 1]` to represent both directions?

Reply:
I believe not.  It looks like in the UI for alerts creation you are only allowed to click `Both Directions`, `Eastbound`, or `Westbound` (eastbound and westbound are route dependent wordings).  You can't select stops from both directions (IE I can't create an alert that affects stop 1 in direction 0, and stop 2 in direction 1.  That is 2 seperate alerts.  So the array should never have more than one direction id in it.  My testing indicates that dotcom transforms both direction alerts are of the form `[null]`.  The rest of this is just to cover bases.